templates/server-config-configmap.yaml

```diff
@@ -14,6 +14,9 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 data:
   extraconfig-from-values.hcl: |-
+  {{- if or (eq .mode "ha") (eq .mode "standalone")}}
+  {{- $type := typeOf (index .Values.server .mode).config }}
```

Comment:
There's actually a new `ha` submode, `raft`, which has it's own configuration.  I think we'll need additional logic for that config file.

In the future we hope to consolidate these modes but we'll need to change this for now.

Reply:
Sorry, I didn't realize about the new submode. My last commit should address this problem.

In a side note, I think that you are over complicating the values file addressing problems where shouldn't be addressed.

It would be extremely more easy to define a single `config` variable with the specifics. If you want to provide examples, do so by creating multiple values files that the users can choose from or simply add comments. That would simplify the code a lot.

The only difference between modes, apart from the config, is the `disruptionBudget` in the `ha`, which could go outside the mode and be used if defined.

Reply:
Hi @davidmogar, thanks for adding this.

As I mentioned in my last comment:

> In the future we hope to consolidate these modes but we'll need to change this for now.

Testing this now and looking to get it merged today.